src/pages/index.js

```diff
@@ -61,66 +76,124 @@ const addPlacePopup = new PopupWithForm(
-  jobSelector: profileJobSelector,
-});
+const imagePopup = new PopupWithImage(imageModalSelector);
+imagePopup.setEventListeners();
+
+const deleteCardPopup = new PopupWithConfirmation(
+  deleteCardModalSelector,
+  deleteCard
+);
+deleteCardPopup.setEventListeners();
 
 function editProfile() {
-  const profileInfo = userInfo.getUserInfo();
-  const { name, job } = profileInfo;
+  const { name, job } = userInfo.getUserInfo();
   nameInput.value = name;
   jobInput.value = job;
   profilePopup.open();
 }
 
 function handleProfileFormSubmit({ name, job }) {
-  userInfo.setUserInfo({ name, job });
-  profilePopup.close();
-  formValidators["profile-form"].resetValidation();
+  profilePopup.setButtonText("Saving...");
+  api
+    .editUserInfo({ name, job })
+    .then(() => {
+      userInfo.setUserInfo({ name, job });
+      profilePopup.close();
+    })
+    .catch((err) => {
+      console.error("Error updating profile:", err);
+    })
+    .finally(() => {
+      profilePopup.setButtonText("Save");
+      formValidators["profile-form"].resetValidation();
+    });
+}
+
+function addPlace() {
+  formValidators["place-form"].resetValidation();
+  addPlacePopup.open();
 }
 
 function handleAddPlaceFormSubmit({ title, url }) {
+  addPlacePopup.setButtonText("Saving...");
   const newCard = { name: title, link: url };
-  const cardElement = getCardElement(newCard);
-
-  cardsList.addItem(cardElement, { method: "prepend" });
-
-  addPlacePopup.close();
+  api
+    .addCard(newCard)
+    .then((savedCard) => {
+      cardsList.renderItems([savedCard]);
+      addPlacePopup.close();
+    })
+    .catch((err) => {
+      console.error("Error adding place:", err);
+    })
+    .finally(() => {
+      addPlacePopup.setButtonText("Create");
+      formValidators["place-form"].toggleButtonState();
+    });
+}
 
-  formValidators["place-form"].toggleButtonState();
+function editAvatar() {
+  const { avatarUrl } = userInfo.getUserInfo();
+  avatarUrlInput.value = avatarUrl;
+  editAvatarPopup.open();
 }
 
-const imagePopup = new PopupWithImage(imageModalSelector);
-imagePopup.setEventListeners();
+function handleEditAvatarSubmit({ url }) {
+  editAvatarPopup.setButtonText("Saving...");
+  api
+    .editAvatarUrl({ url })
+    .then(() => {
+      userInfo.setAvatar({ avatarUrl: url });
+      editAvatarPopup.close();
+    })
+    .catch((err) => {
+      console.error("Error updating avatar:", err);
+    })
+    .finally(() => {
+      editAvatarPopup.setButtonText("Save");
+      formValidators["edit-avatar-form"].resetValidation();
+    });
+}
 
 function handleImageClick(data) {
   imagePopup.open(data);
 }
 
-editProfileButton.addEventListener("click", editProfile);
-addPlaceButton.addEventListener("click", addPlace);
+function openDeleteCardPopup({ cardId }) {
+  deleteCardPopup.setCardId(cardId);
+  deleteCardPopup.open();
+}
 
-const formValidators = {};
+function deleteCard(cardId) {
+  document.querySelector(`#${cardId}`).remove();
```

Comment:
You need to **delete** cards only in a `then` block after a **successful** response from the server, so that if there is an error, the card should stay in the `DOM`.